rts/lra/lra-proxy/api/src/main/java/io/narayana/lra/client/internal/proxy/nonjaxrs/LRAParticipant.java

```diff
@@ -111,8 +110,10 @@ synchronized Response forget(URI lraId, URI parentId) {
         return invokeParticipantMethod(forgetMethod, lraId, parentId, FORGET);
     }
 
-    synchronized void afterLRA(URI lraId, LRAStatus lraStatus) {
+    synchronized Response afterLRA(URI lraId, LRAStatus lraStatus) {
         invokeMethod(AFTER, afterLRAMethod, getInstance(), lraId, lraStatus);
+        // above invocation throws RuntimeException in case an unexpected exception is thrown from client method
+        return Response.ok().build();
```

Comment:
If the invokeMethod call fails then the implementation needs to know so that it can keep retrying the afterLRA notification.

Reply:
RuntimeException translates to 500 internal server error, so the coordinator will retry the call.

Please note that this will interpret WebApplicationException (as in other participant methods invocations) so if the user includes 200 status code in this exception it will be interpreted as a successful call. Is that a problem?

Reply:
You have some code in invokeMethod which will returns processThrowable(e.getTargetException(), type) which returns an HTTP status code if the target method throws an exception. But your afterLRA method ignores that return value.